### PR TITLE
Document how to update a "develop branch to develop branch" pull request

### LINKED ISSUE #9508: Document how to update a "develop branch to develop branch" pull request
Originally posted at https://github.com/IQSS/dataverse/pull/8970#issuecomment-1430241288 below is how I've successfully merged the latest from IQSS "develop" branch into the pull request's "develop" branch.

We should document this (as well as discourage contributors from doing this).

```
# do all this in /tmp away from your normal code
cd /tmp
git clone git@github.com:IQSS/dataverse.git
cd dataverse
git remote add noobyu6 git@github.com:noobyu6/dataverse.git
git fetch noobyu6
# make a new branch not called develop (dev2)
git checkout noobyu6/develop -b dev2
git merge origin/develop
# delete the real develop branch (!) locally
git branch -d develop
# checkout dev2 (false develop)  as develop for now
git checkout -b develop
# push the false develop to the contributor's fork (to the PR)
git push noobyu6 develop 
cd ..
# delete the tmp space
rm -rf /tmp/dataverse
```

### PR DESCRIPTION
**Pull request by martinAmouzou**
**What this PR does / why we need it**: This PR is to solve the issue 9508 on how to "update a develop branch to develop branch" pull request

**Which issue(s) this PR closes**:

- Closes #9508 

**Special notes for your reviewer**:

**Suggestions on how to test this**: The modifications are on the latest section of the doc (version-control.rst)

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**: No

**Additional documentation**:

Preview at https://dataverse-guide--10261.org.readthedocs.build/en/10261/developers/version-control.html#handing-a-pull-request-from-a-develop-branch

### CODE REVIEW

File: doc/sphinx-guides/source/developers/version-control.rst [hunk 1]
```diff
@@ -287,4 +287,8 @@ GitHub documents how to make changes to a fork at https://help.github.com/articl
 
 ----
 
+Develop branch and pull request 
+-------------------------------------------- 
+Please do not use your forked develop branch to push a PR and follow the `1st scenario: preparing the first pull request of Version Control Guide <https://github.com/Recherche-Data-Gouv/dataverse/blob/develop/doc/sphinx-guides/source/developers/version-control.rst>`_.
```
Comment (pdurbin) [inline doc/sphinx-guides/source/developers/version-control.rst:292]: I'm confused by this link. We're already in version-control.rst. Why would we link to the file we are already in? 🤔

**Review by pdurbin — COMMENTED** [review]
Just a quick pass... thanks for the pull request! ❤️

File: doc/sphinx-guides/source/developers/version-control.rst (hunk 1 above)
Comment (martinAmouzou) [reply doc/sphinx-guides/source/developers/version-control.rst]: I redirected to the section on git commands. I just corrected the link.

Comment (pdurbin) [reply doc/sphinx-guides/source/developers/version-control.rst]: @martinAmouzou ah, thanks, I understand now and that `#` thing is a good sphinx/rst trick!

Is it ok if I move what you wrote somewhere else? And add a bit more content (for maintainers like me)?

Comment (martinAmouzou) [reply doc/sphinx-guides/source/developers/version-control.rst]: @pdurbin Yeah, you can do so. thanks

Comment (pdurbin) [reply doc/sphinx-guides/source/developers/version-control.rst]: Great, thanks!

**Review by pdurbin — APPROVED** [review]
@martinAmouzou I wrote what I had in mind and will go ahead and merge this. Thanks!

**Pull request merged**